### PR TITLE
Add pillar printing - Rebase of #351

### PR DESCRIPTION
**Pull request by gergness**
Rebase of PR to add pillar printing for labelled vectors.

Here's what an example data.frame looks like printed out on my machine.
```r
library(haven)
examples <- dplyr::data_frame(
  x1 = labelled_spss(c(1:10, NA), c(Good = 1, Bad = 8), c(9, 10)), 
  x2 = labelled_spss(c(1:10, NA), c(Good = 1, Bad = 8, Refused = 10), c(9, 10)),
  x3 = labelled(c(1:8, tagged_na("a"), tagged_na("b"), NA), c(Good = 1, "Bad" = 8)),
  x4 = labelled(c(1:8, tagged_na("a"), tagged_na("b"), NA), c(Good = 1, "Bad" = 8, Refused = tagged_na("b"))),
  x5 = labelled(1:11, c("Not used" = 12)),
  x6 = labelled(c(rep(c(1.22352, 1000, -345), each = 3), 35, 35), c(One = 1.22352, Two =  35, Threeeee = 1000)),
  x7 = labelled(c(rep("A", 3), rep("B", 3), rep("XXXXXX", 4), NA), c(Apple = "A", Banana = "B", Mystery = "XXXXXX"))
)

examples
```

screen shot 2018-08-05 at 2 26 58 pm

And here's the same data.frame with various cutoffs for truncation:

```r
print(examples, width = 70)
```
screen shot 2018-08-05 at 2 27 53 pm

```r
print(examples, width = 75)
```
screen shot 2018-08-05 at 2 27 40 pm

```r
print(examples, width = 80)
```
screen shot 2018-08-05 at 2 28 07 pm

### CODE REVIEW

**Comment by gergness** [comment]
Looking closer, the padding  on x6 value label "Threeee" is off by a space. I can take a look at that, but am hoping to hear back from you if you like the general direction before continuing.

**Comment by hadley** [comment]
Coming back to this after a long delay — @gergness are you still interested in finishing this off?

**Comment by gergness** [comment]
Yeah, sounds good! I just brought it up to date with master, but let me know if you'd like any other changes.

File: DESCRIPTION [hunk 1]
```diff
@@ -27,7 +27,10 @@ Imports:
     hms,
     Rcpp (>= 0.11.4),
     readr (>= 0.1.0),
-    tibble
+    tibble,
+    pillar (>= 1.1.1),
```
Comment (hadley) [inline DESCRIPTION]: I think we need to make these suggested packages. Do you know the process for registering methods for suggested packages? (If not, I'm happy to explain it)

Comment (gergness) [reply DESCRIPTION]: Looks like hms is a good template for me to use (and has tests). I'll let you know if I hit any snags though.

File: tests/testthat/test-pillar.R [hunk 2]
```diff
@@ -0,0 +1,46 @@
+context("pillar printing")
+
+test_that("pillar", {
+  skip_if_not_installed("pillar")
```
[the remark this answers is not in the document]
Comment (gergness) [reply tests/testthat/test-pillar.R]: No problem. I remembered [this](http://dirk.eddelbuettel.com/blog/2017/03/22/) coming from you:


(though upon a reread, the tone doesn't quite match your usual style  ;-) )

[the remark this answers is not in the document]
Comment (hadley) [reply tests/testthat/test-pillar.R]: Yeah, that's a belief that we don't subscribe to.

File: tests/testthat/pillar_double.txt [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# A tibble: 11 x 1
+   x               
+   <dbl+lbl>       
+ 1    1.22 [One]
```
Comment (hadley) [inline tests/testthat/pillar_double.txt]: The alignment looks off to me here — the numbers should be aligned along their decimal places

File: tests/testthat/pillar_spss_integer.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+   <int+lbl>       
+ 1  1 [Good]       
+ 2  2              
+ 3  3              
+ 4  4              
+ 5  5              
+ 6  6              
+ 7  7              
+ 8  8 [Bad]        
+ 9  9(NA)
```
Comment (hadley) [inline tests/testthat/pillar_spss_integer.txt]: Maybe space before `(`?

File: tests/testthat/pillar_tagged_na.txt [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+   x              
+   <dbl+lbl>      
+ 1  1 [Good]      
+ 2  2             
+ 3  3             
+ 4  4             
+ 5  5             
+ 6  6             
+ 7  7             
+ 8  8 [Bad]
```
Comment (hadley) [inline tests/testthat/pillar_tagged_na.txt]: I think the numbers should be right aligned to the `)` in the tagged NAs

**Comment by gergness** [comment]
Screen shot to show colors: 
image

**Comment by hadley** [comment]
Thanks for all your work on this!

**Pull request merged**